src/main/java/pvs/app/controller/GitlabApiController.java

```diff
@@ -0,0 +1,119 @@
+package pvs.app.controller;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import lombok.SneakyThrows;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+import org.gitlab4j.api.GitLabApiException;
+import pvs.app.dto.GitlabCommitDTO;
+import pvs.app.dto.GitlabIssueDTO;
+import pvs.app.service.GitlabApiService;
+import pvs.app.service.GitlabCommitService;
+
+import java.io.IOException;
+import java.util.Calendar;
+import java.util.Date;
+import java.util.List;
+
+@RestController
+@RequestMapping(produces = MediaType.APPLICATION_JSON_VALUE)
+public class GitlabApiController {
+
+    static final Logger logger = LogManager.getLogger(GithubApiController.class.getName());
+    private final GitlabApiService gitlabApiService;
+    private final GitlabCommitService gitlabCommitService;
+    @Value("${message.exception}")
+    private String exceptionMessage;
+
+    public GitlabApiController(GitlabApiService gitlabApiService, GitlabCommitService gitlabCommitService){
+        this.gitlabApiService = gitlabApiService;
+        this.gitlabCommitService = gitlabCommitService;
+    }
+
+    @SneakyThrows
+    @GetMapping("/gitlab/commits/{repoOwner}/{repoName}")
+    public ResponseEntity<String> getCommits(@PathVariable("repoOwner") String repoOwner, @PathVariable("repoName") String repoName) throws GitLabApiException {
+        ObjectMapper objectMapper = new ObjectMapper();
+        List<GitlabCommitDTO> gitlabCommitDTOs = gitlabCommitService.getAllCommits(repoOwner, repoName);
+        String gitlabCommitDTOsJson;
+
+        try {
+            gitlabCommitDTOsJson = objectMapper.writeValueAsString(gitlabCommitDTOs);
```

Comment:
The `gitlabCommitDTOsJson` is only use in the `try { ... }` scope, so suggest move it into that scope.

```suggestion

        try {
            String gitlabCommitDTOsJson = objectMapper.writeValueAsString(gitlabCommitDTOs);
```